chunk/chunk.py

```diff
@@ -101,65 +146,74 @@ def write_bytes_to_target(target_uri, contents):
-        source_uri = vsi_curlify(uri)
-
-        # Get the workspace 
-        workspace_key = get_filename(uri)
-        i = 1
-        while workspace_key in workspace_keys:
-            if i > 2:
-                workspace_key = workspace_key[:-2] + "-" + str(i)
-            else:
-                workspace_key = workspace_key + "-" + str(i)
-            i + 1
-        workspace_keys.append(workspace_key)
-        
-        workspace_target = os.path.join(workspace_uri, workspace_key + "-workingcopy.tif")
-        workspace_source_uri = vsi_curlify(workspace_target)
-
-        image_folder = os.path.join(workspace_uri, workspace_key)
-        
-        uri_set = UriSet(source_uri = source_uri,
-                         workspace_target = workspace_target,
-                         workspace_source_uri = workspace_source_uri,
-                         image_folder = image_folder,
-                         order = order)
-                         
-        result.append(uri_set)
-
-    return result
-
-def copy_to_workspace(source_uri, dest_uri):
-    """
-    Translates an image from a URI to a compressed, tiled GeoTIFF version in the workspace
-    """
-
-    creation_options = {
-        "driver": "GTiff",
-        "tiled": True,
-        "compress": "lzw",
-        "predictor":   2,
-        "sparse_ok": True,
-        "blockxsize": 512, 
-        "blockysize": 512
-    }
-
-    with rasterio.open(source_uri, "r") as src:
-        meta = src.meta.copy()
-        meta.update(creation_options)
-
-        tmp_path = "/vsimem/" + get_filename(dest_uri)
-
-        with rasterio.open(tmp_path, "w", **meta) as tmp:
-            tmp.write(src.read())
-
-    contents = bytearray(virtual_file_to_buffer(tmp_path))
-
-    write_bytes_to_target(dest_uri, contents)
+def copy_tiles_to_workspace(source_uri, order, workspace_uri):
+    # Download the file and retile
+    results = []
+    workspace_prefix = get_filename(source_uri)
+    
+    local_dir = create_tmp_directory(workspace_prefix)
+    try :
+        MAX_HEIGHT = 1024 * 2
+        MAX_WIDTH = 1024 * 2
+
+        local_path = get_local_copy(source_uri, local_dir)
+
+        gdal_options = ["-of", "GTiff",
+                        "-co", "compress=lzw",
+                        "-co", "predictor=2",
+                        "-co", "tiled=yes",
+                        "-co", "blockxsize=512",
+                        "-co", "blockysize=512"]
+
+        # reproject
+        reprojected_path = local_path + "-reprojected.tif"
+        cmd = ["gdalwarp"] + gdal_options + ["-t_srs", "EPSG:3857",
+                                             "-co", "BIGTIFF=YES", # Handle giant TIFFs.
+                                             local_path,
+                                             reprojected_path]
+        call(cmd)
+
+        # retile
+        tiled_dir = local_path + "-tiled"
+        os.mkdir(tiled_dir)
+        cmd = ["gdal_retile.py"] + gdal_options + ["-ps", 
```

Comment:
True, it's what we'd want without the distributed windowed nature (which relies on the rasterio-disapproved vsicurl), and without reprojecting to web mercator all in one swoop. Beforehand we were translating the images to GeoTiffs in their native CRS and doing the WM tile creation as the windowed read via vsicurl. This does the reproject and the tiling all on one machine, which is slower but doesn't segfault on large images.